src/toxcore/dht/dht_node.rs

```diff
@@ -154,13 +154,44 @@ impl DhtNode {
                         None => Some(SocketAddr::V6(v6)),
                     }
                 },
-                None => {
-                    self.assoc4.saddr.map(Into::into)
-                },
+                None => self.assoc4.saddr.map(Into::into),
             }
         } else {
             self.assoc4.saddr.map(Into::into)
+        };
+
+        if addr.is_none() {
+            warn!("get_socket_addr: failed to get address of DhtNode");
         }
+
+        addr
+    }
+
+    /// Returns all available socket addresses of DhtNode
+    pub fn get_all_sockets(&self, is_ipv6_enabled: bool) -> Vec<SocketAddr> {
+        let mut addrs = Vec::new();
+
+        if is_ipv6_enabled {// DHT node is running in ipv6 mode
+            self.assoc6.saddr.and_then(|v6| Some(addrs.push(SocketAddr::V6(v6))));
```

Comment:
`if let` is better here since you don't care of the result of `and_then` and mutate `addrs` inside.